lib/processing/file_processing.ex

```diff
@@ -6,11 +6,11 @@ defmodule Uploadex.FileProcessing do
   @doc """
   If it's in base64, decode it. Otherwise, do not try to process the file.
   """
-  @spec process_binary(String.t) :: {:ok, binary()} | :error | {:error, keyword()}
+  @spec process_binary(String.t) :: {String.t, {:ok, binary()}} | :error | {:error, keyword() | String.t}
```

Comment:
Should be

```
@spec process_binary(String.t) :: {String.t, {:ok, binary()}} | {:error, String.t} | {String.t, {:error, keyword}}
```

Or maybe handle this to always return `{:error, keyword | String.t}`?

`{:ok, %{content_type: "", binary: <<>>}` and `{:error, error}` would be a nicer API